send-report.sh

```diff
@@ -0,0 +1,209 @@
+#!/bin/bash -e
+
+BUNDLE_TEMPLATE="transaction-bundle-template.json"
+
+usage() {
+    echo "Usage: $0 <report-file> <report-server> <author-ident-sys> <author-ident-val> <proj-ident-sys> <proj-ident-val>"
+    echo "       $0 <report-file> <report-server> <author-ident-sys> <author-ident-val> <proj-ident-sys> <proj-ident-val> [-u <user> -p <password>]"
+    echo "       $0 <report-file> <report-server> <author-ident-sys> <author-ident-val> <proj-ident-sys> <proj-ident-val> [-i <issuer-url> -c <client-id> -s <client-secret>]"
+    exit 1
+}
+
+if [[ $# -lt 6 ]]; then
+    usage
+fi
+
+report_file="$1"
+report_server="$2"
+author_ident_sys="$3"
+author_ident_val="$4"
+proj_ident_sys="$5"
+proj_ident_val="$6"
+
+user=""
+password=""
+issuer_url=""
+client_id=""
+client_secret=""
+
+
+read_http_args() {
+
+    while getopts ":u:p:i:c:s:" opt; do
+            case $opt in
+                u)
+                    user="$OPTARG"
+                    ;;
+                p)
+                    password="$OPTARG"
+                    ;;
+                i)
+                    issuer_url="$OPTARG"
+                    ;;
+                c)
+                    client_id="$OPTARG"
+                    ;;
+                s)
+                    client_secret="$OPTARG"
+                    ;;
+                *)
+                    echo "Unknown option: -$OPTARG"
+                    usage
+                    ;;
+            esac
+        done
+}
+
+# This is needed to see if there is already a DocumentReference with the given project identifier.
+# In that case, the id of the DocumentReference is later used to update the DocumentReference.
+http_get_all_doc_refs() {
+    auth="$1"
+    if [ "$auth" == "no-auth" ]; then
+        response=$(curl -s "$report_server/DocumentReference" \
+                        -H "Content-Type: application/fhir+json")
+        echo "$response"
+    elif [ "$auth" == "basic-auth" ]; then
+        response=$(curl -s "$report_server/DocumentReference" \
+                        -H "Content-Type: application/fhir+json" \
+                        -u "$user:$password")
+        echo "$response"
+    else
+        oauth_response=$(curl -s -X POST "$issuer_url" \
+                            -H "Content-Type: application/x-www-form-urlencoded" \
+                            -d "grant_type=client_credentials" \
+                            -d "client_id=$client_id" \
+                            -d "client_secret=$client_secret")
+        fhir_report_bearer_token=$(echo "$oauth_response" | jq -r '.access_token')
+
+        response=$(curl -s "$report_server/DocumentReference" \
+                    -H "Content-Type: application/fhir+json" \
+                    -H "Authorization: Bearer $fhir_report_bearer_token")
```

Comment:
There is the Flag `--oauth2-bearer` that just takes the token and creates the header.